private/pkg/app/appcmd/webpages.go

```diff
@@ -0,0 +1,305 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package appdoc provides a markdown generator for cobra commands.
+// In the future this will need to be adapted to appcmd.Command when we remove usage of Cobra.
+package appcmd
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"html"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"unicode"
+
+	"github.com/bufbuild/buf/private/pkg/app"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var codeBlockRegex = regexp.MustCompile(`(^\s\s\s\s)|(^\t)`)
+
+const (
+	slugPrefixFlagName      = "slug-prefix"
+	excludeCommandsFlagName = "exclude-command"
+)
+
+type flags struct {
+	SlugPrefix      string
+	ExcludeCommands []string
+}
+
+func newFlags() *flags {
+	return &flags{}
+}
+
+func (f *flags) Bind(flagSet *pflag.FlagSet) {
+	flagSet.StringVar(
+		&f.SlugPrefix,
+		slugPrefixFlagName,
+		"",
+		"The slug prefix for front-matter slug attribute",
+	)
+	flagSet.StringSliceVar(
+		&f.ExcludeCommands,
+		excludeCommandsFlagName,
+		nil,
+		"Exclude these commands from doc generation",
+	)
+}
+
+// newWebpagesCommand returns a new Command.
+func newWebpagesCommand(
+	command *cobra.Command,
+) *Command {
+	flags := newFlags()
+	return &Command{
+		Use:    "webpages",
+		Args:   cobra.ExactArgs(1),
+		Hidden: true,
+		Run: func(ctx context.Context, container app.Container) error {
+			excludes := make(map[string]bool)
+			for _, exclude := range flags.ExcludeCommands {
+				excludes[exclude] = true
+			}
+			for _, cmd := range command.Commands() {
+				if excludes[cmd.CommandPath()] {
+					cmd.Hidden = true
+				}
+			}
+			return generateMarkdownTree(
+				command,
+				container.Arg(0),
+				flags.SlugPrefix,
+			)
+		},
+		BindFlags: flags.Bind,
+	}
+}
+
+// generateMarkdownTree generates markdown for a whole command tree.
+func generateMarkdownTree(cmd *cobra.Command, dir string, slugPrefix string) error {
+	if !cmd.IsAvailableCommand() {
+		return nil
+	}
+	for _, c := range cmd.Commands() {
```

Comment:
for _, command := range commands

Reply:
Resolved: 04234f92c37802cad9fc767a4ed184dfad84ff31